inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/convert_mul_or_add_finally.cpp

```diff
@@ -3,11 +3,295 @@
+                auto biases_in = ngraph::op::util::normalize_constant(bias, output_shape);
+                if (is_dequantization) {
+                    const ngraph::Shape data_shape = data_node.get_shape();
+                    ngraph::Shape broadcasted_shape = std::vector<size_t>(data_shape.size(), 1ul);
+                    broadcasted_shape[1] = data_shape[1];
+
+                    weights_in = ngraph::op::util::broadcastTo(weights_in, broadcasted_shape);
+                    biases_in = ngraph::op::util::broadcastTo(biases_in, broadcasted_shape);
+                }
+                scaleshift = std::make_shared<ngraph::op::ScaleShiftIE>(data_node, weights_in, biases_in);
+            } else {
+                return false;
+            }
+
+            scaleshift->set_friendly_name(lin_op->get_friendly_name());
+            ngraph::copy_runtime_info(m.get_match_root(), scaleshift);
+            ngraph::replace_node(m.get_match_root(), scaleshift);
+        } else {
+            float value;
+            if (!ngraph::op::util::get_single_value(const_node, value)) {
+                return false;
+            }
+
+            // In case Add we create fake scale equal to 1, in case of Multiply we create fake shift equal to 0
+            std::shared_ptr<ngraph::op::PowerIE> power;
+            if (std::is_same<T, ngraph::opset1::Add>()) {
+                power = std::make_shared<ngraph::op::PowerIE>(data_node, 1.0f, 1.0f, value, lin_op->get_output_element_type(0));
+            } else if (std::is_same<T, ngraph::opset1::Multiply>()) {
+                power = std::make_shared<ngraph::op::PowerIE>(data_node, 1.0f, value, 0.0f, lin_op->get_output_element_type(0));
+            } else if (std::is_same<T, ngraph::opset1::Subtract>()) {
+                power = std::make_shared<ngraph::op::PowerIE>(data_node, 1.0f, 1.0f, -value, lin_op->get_output_element_type(0));
+            } else {
+                return false;
+            }
+            power->set_friendly_name(lin_op->get_friendly_name());
+            ngraph::copy_runtime_info(m.get_match_root(), power);
+            ngraph::replace_node(m.get_match_root(), power);
+        }
+
+        return true;
+    };
+    return callback;
+}
+
+class ConvertAdd: public ngraph::pass::MatcherPass {
+public:
+    ConvertAdd() {
+        auto m = std::make_shared<ngraph::pattern::Matcher>(ngraph::pattern::wrap_type<ngraph::opset1::Add>());
+        register_matcher(m, get_callback<ngraph::opset1::Add>());
+    }
+};
+
+class ConvertSub: public ngraph::pass::MatcherPass {
+public:
+    ConvertSub() {
+        auto m = std::make_shared<ngraph::pattern::Matcher>(ngraph::pattern::wrap_type<ngraph::opset1::Subtract>());
+        register_matcher(m, get_callback<ngraph::opset1::Subtract>());
+    }
+};
+
+class ConvertMul: public ngraph::pass::MatcherPass {
+public:
+    ConvertMul() {
+        auto m = std::make_shared<ngraph::pattern::Matcher>(ngraph::pattern::wrap_type<ngraph::opset1::Multiply>());
```

Comment:
Please add CC macros. You can find examples in the PR https://github.com/openvinotoolkit/openvino/pull/3795

Reply:
@ilyachur looks like we don't have `itt.hpp` with `MATCHER_PASS_SCOPE` macro for legacy_library which I think also need to be updated with itt. Could you handle it in a separate PR?

Reply:
Oh yes, I thought it a transformation library. 
Yes we can handle it separately